ament_cmake_core/cmake/index/ament_index_get_resources.cmake

```diff
@@ -19,7 +19,20 @@ function(ament_index_get_resources var resource_type)
       "arguments: ${ARGN}")
   endif()
 
-  string(REPLACE ":" ";" paths_to_search "$ENV{AMENT_PREFIX_PATH}")
+  if(NOT WIN32)
+    string(REPLACE ":" ";" raw_paths_to_search "$ENV{AMENT_PREFIX_PATH}")
+  else()
+    # The path separator is already ; on Windows, plus the C:\'s : would match.
+    set(raw_paths_to_search "$ENV{AMENT_PREFIX_PATH}")
+  endif()
```

Comment:
Would the following be more readable?

    set(raw_paths_to_search "$ENV{AMENT_PREFIX_PATH}")
    if(NOT WIN32)
        # convert the path separator to CMake list separators
        string(REPLACE ":" ";" raw_paths_to_search "${raw_paths_to_search}")
    endif()